FWCore/Services/web/transitions.js

```diff
@@ -0,0 +1,496 @@
+	"end process block",
+	0, //unused -7
+	"write end run", 
+	"global end run", //global end run
+	"stream end run", //stream end run
+	"write end lumi", //write end lumi
+	"global end lumi", //global end lumi
+	"stream end lumi", //stream end lumi
+	"event", //event
+	"stream begin lumi", //stream begin lumi
+	"global begin lumi", //global begin lumi
+	0, //unused 3
+	"stream begin run", //stream begin run
+	"global begin run", //global begin run
+	0, //unused 6
+	"access input process block", //access input process block
+	"begin process block", //begin process block
+	"open file",
+	"begin stream", //begin stream
+	"begin job", //begin job
+	"EventSetup synch", //sync
+	"EventSetup sych enqueue", //sync enqueue
+	"construction", //construction
+	"startup", //startup 
+]
+
+const activityToName = [  "prefetch",
+  "acquire",
+  "process", 
+  "delayedGet"]
+
+const activityToColor = ["#FF5F1F", "#CC7722", null, "#FF4433"];
+
+window.onload = () => {
+
+//const response = await fetch("./data.json");
+//const data = await response.json();
+const bar = document.querySelector('.split__bar');
+const left = document.querySelector('.name_div');
+const div = document.querySelector('.graph_div');
+const bottom = document.querySelector('.time_div');
+const graph = document.getElementById('graph_view');
+const graph_context = graph.getContext('2d');
+const name_view = document.getElementById('name_view');
+const name_context = name_view.getContext('2d');
+const time_view = document.getElementById('time_view');
+const time_context = time_view.getContext('2d');
+const selected_view = document.getElementById('selected_paragraph');
+let selected_item = null;
+selected_view.innerHTML = "Selected: [click on box in graph]";
+let mouse_is_down = false;
+
+const max_graph_height = 500;
+
+//same scale as window coordinates
+let graph_origin = {x: 0, y: 0};
+let timeZoomFactor = 1.0;
+
+bar.addEventListener('mousedown', (e) => {
+  mouse_is_down = true;
+})
+
+document.addEventListener('mousemove', (e) => {
+  if (!mouse_is_down) return;
+
+  left.style.width = `${e.clientX}px`;
+})
+
+document.addEventListener('mouseup', () => {
+  mouse_is_down = false;
+})
+
+function maxTime() {
+	let maxTime = 0;
+	for( let grouping of data.transitions) {
+		for( let slot of grouping.slots) {
+			for (let transition of slot) {
+				if (maxTime < transition.finish) {
+					maxTime = transition.finish;
+				}
+			}
+		}
+	}
+
+	maxTime = Math.ceil(maxTime);
+	
+	const digits = maxTime.toString().length;
+	const digitsToZeroOut = digits -2;
+	if (digitsToZeroOut > 0) {
+		for(let i = 0; i < digitsToZeroOut; ++i) {
+			maxTime /= 10;
+		}
+		maxTime *=10;
+		maxTime +=9;
+		for( let i = 1; i < digitsToZeroOut; ++i) {
+			maxTime *=10;
+		}
+		maxTime = Math.ceil(maxTime);
+	}
+	return maxTime;
+}
+
+const kEndTime = maxTime();
+
+function drawNames() {
+    name_context.save()
+    name_context.setTransform(1,0,0,1,0,0);
+	name_context.fillStyle = "#AACD6E"
```

Comment:
Indentation seems to be inconsistent on some lines (some have tabs, others have spaces)

Reply:
reindented